### ISSUE TITLE
cmd/go: go list -compiled -e fails completely if any package does not compile

### ISSUE THREAD

**Issue by matloob** [golang/go#26755]
This is on tip. go version: go version devel +b8669ef1ce Wed Aug 1 14:29:58 2018 +0000 linux/amd64

go list -e should not fail if there are broken packages, or print errors to standard error, as per the documentation:
```
The -e flag changes the handling of erroneous packages, those that
cannot be found or are malformed. By default, the list command
prints an error to standard error for each erroneous package and
omits the packages from consideration during the usual printing.
With the -e flag, the list command never prints errors to standard
error and instead processes the erroneous packages with the usual
printing. Erroneous packages will have a non-empty ImportPath and
a non-nil Error field; other information may or may not be missing
(zeroed).
```
go list -compiled -e breaks this.

Repro:

```
matloob@matloob:~$ mkdir /tmp/empty
matloob@matloob:~$ cd /tmp/empty
matloob@matloob:/tmp/empty$ devgo version
go version devel +b8669ef1ce Wed Aug 1 14:29:58 2018 +0000 linux/amd64
matloob@matloob:/tmp/empty$ devgo list -e -compiled >/dev/null
go build _/tmp/empty: no Go files in /tmp/empty
matloob@matloob:/tmp/empty$ echo $?
1
```

**Comment by myitcv** [comment]
Dup of https://github.com/golang/go/issues/25842?

**Comment by bcmills** [comment]
Probably a duplicate, but it wouldn't hurt to test `-export` and `-compiled` separately.

**Comment by matloob** [comment]
Do we think this'll get fixed for 1.13?

**Comment by bcmills** [comment]
Not sure. Only if the fix is simple, but I suspect it will not be.

**Comment by nabice** [comment]
I think stdout should be returned, even though `go list -e` returns a non-zero exit.
It will be helpful to collect more information for go/analysis.
https://github.com/golang/tools/blob/38ae2c8f64122bd595b7f93f968a6686cd27bb5a/go/packages/golist.go#L802-L816

**Comment by agnivade** [comment]
This is what I get on tip (+57ad6ef15d Tue Oct 22 04:05:25 2019 +0000 linux/amd64) now

```
12:29:25-agniva-/tmp/empty$gotip list -e -compiled > /dev/null 
12:29:33-agniva-/tmp/empty$echo $?
0
```

@bcmills ?

**Comment by bcmills** [comment]
The original reproducer does seem to be fixed in 1.14.

```
$ mkdir empty

$ cd empty

empty$ go1.14beta1 list -e -compiled
_/tmp/tmp.xl7JwKGEfx/empty

empty$ GO111MODULE=on go1.14beta1 list -e -compiled
go: cannot find main module; see 'go help modules'

empty$ GO111MODULE=on go1.14beta1 mod init empty
go: creating new go.mod: module empty

empty$ GO111MODULE=on go1.14beta1 list -e -compiled
_/tmp/tmp.xl7JwKGEfx/empty

empty$
```

@matloob, do you have any other reproducers for this?
